client/landing/stepper/declarative-flow/internals/steps-repository/sensei-plan/index.tsx

```diff
@@ -0,0 +1,396 @@
+		select( ONBOARD_STORE )
+	);
+	const domain = getSelectedDomain();
+	const { getNewSite } = useSelect( ( select ) => select( SITE_STORE ) );
+	const { getPlanProduct } = useSelect( ( select ) => select( PLANS_STORE ) );
+
+	const { saveSiteSettings, setIntentOnSite } = useDispatch( SITE_STORE );
+
+	const planObject = supportedPlans.find( ( plan ) => {
+		return plan && 'business' === plan.periodAgnosticSlug;
+	} );
+
+	const handlePlanBillingPeriodChange = ( newBillingPeriod: Plans.PlanBillingPeriod ) => {
+		setBillingPeriod( newBillingPeriod );
+	};
+
+	const woothemesProduct = useSelect(
+		( select ) => {
+			const yearly = select( PRODUCTS_LIST_STORE ).getProductBySlug( 'woothemes_sensei_yearly' );
+			const monthly = select( PRODUCTS_LIST_STORE ).getProductBySlug( 'woothemes_sensei_monthly' );
+
+			if ( ! yearly || ! monthly ) {
+				return {
+					monthlyPrice: 0,
+					yearlyPrice: 0,
+					price: 0,
+					productSlug: '',
+					currencyCode: '',
+				};
+			}
+
+			return {
+				monthlyPrice: monthly.cost,
+				yearlyPrice: yearly.cost,
+				price: billingPeriodIsAnnually ? Math.ceil( yearly.cost / 12 ) : monthly.cost,
+				productSlug: billingPeriodIsAnnually ? yearly.product_slug : monthly.product_slug,
+				currencyCode: yearly.currency_code,
+			};
+		},
+		[ billingPeriodIsAnnually ]
+	);
+
+	const planProduct = useSelect(
+		( select ) => {
+			const monthly = select( PLANS_STORE ).getPlanProduct(
+				planObject?.periodAgnosticSlug,
+				'MONTHLY'
+			);
+			const yearly = select( PLANS_STORE ).getPlanProduct(
+				planObject?.periodAgnosticSlug,
+				'ANNUALLY'
+			);
+			if ( ! yearly || ! monthly ) {
+				return {
+					monthlyPrice: 0,
+					yearlyPrice: 0,
+					price: 0,
+					productSlug: '',
+				};
+			}
+
+			return {
+				monthlyPrice: monthly.rawPrice,
+				yearlyPrice: yearly.rawPrice,
+				price: billingPeriodIsAnnually ? Math.ceil( yearly.rawPrice / 12 ) : monthly.rawPrice,
+				productSlug: '',
+			};
+		},
+		[ billingPeriodIsAnnually, planObject ]
+	);
+
+	const goToDomainStep = useCallback( () => {
+		if ( goToStep ) {
+			goToStep( 'senseiDomain' );
+		}
+	}, [ goToStep ] );
+
+	const onPlanSelect = async () => {
+		try {
+			setStatus( Status.Bundling );
+
+			// Wait for a bit to get an animation in the beginning of site creation.
+			await new Promise( ( res ) => setTimeout( res, 100 ) );
+
+			setProgress( {
+				percentage: 25,
+				title: siteProgressTitle,
+			} );
+
+			await createSenseiSite( {
+				username: currentUser?.username || '',
+				languageSlug: locale,
+				visibility,
+			} );
+
+			setProgress( {
+				percentage: 50,
+				title: cartProgressTitle,
+			} );
+
+			const newSite = getNewSite();
+			const siteId = newSite?.blogid;
+			setSelectedSite( newSite?.blogid );
+			await setIntentOnSite( newSite?.site_slug as string, SENSEI_FLOW );
+			await saveSiteSettings( newSite?.blogid as number, { launchpad_screen: 'full' } );
```

Comment:
Since not handling errors, let's run them in parallel for better perf.

```suggestion
			await Promise.all([
				setIntentOnSite( newSite?.site_slug as string, SENSEI_FLOW ),
				saveSiteSettings( newSite?.blogid as number, { launchpad_screen: 'full' })
			]
```